src/libos/src/fs/file.rs

```diff
@@ -67,7 +68,7 @@ pub trait File: Debug + Sync + Send + Any {
         Ok(())
     }
 
-    fn ioctl(&self, cmd: &mut IoctlCmd) -> Result<i32> {
+    fn ioctl(&self, cmd: &mut IoctlCmd, entry: &mut FileTableEntry) -> Result<i32> {
```

Comment:
I think this implementation is not elegant, because you add a FileTableEntry into the input of ioctl. I think we can handle the FIONCLEX and FIOCLEX  through the `do_ioctl` function in the`src/fs/file_ops/ioctl/mod.rs` for all file entries.

Reply:
Good suggestions. Done. Thank you for your reivew.